enhancements/authentication/configuring-webhook-token-authenticators.md

```diff
@@ -0,0 +1,159 @@
+---
+title: configuring-webhook-token-authenticators
+authors:
+  - "@stlaz"
+reviewers:
+  - "@sttts"
+  - "@deads2k"
+approvers:
+  - "@deads2k"
+  - "@sttts"
+  - "@mfojtik"
+creation-date: 2019-12-10
+last-updated: 2019-12-10
+status: implementable
+see-also:
+  - ""
+replaces:
+  - ""
+superseded-by:
+  - ""
+---
+
+# Configuring Webhook Token Authenticators
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+  1. Should we leverage the `webhookTokenAuthenticators` field of the `authentications.config.openshift.io`
+     resource or should we use CRDs instead?
+  3. Will any changes be needed for the console?
+  4. We add `system:authenticated:oauth` group ([link to code](https://github.com/openshift/origin/blob/7540279a861809a1df1fde4bd94143ba042f0bd6/vendor/k8s.io/kubernetes/pkg/kubeapiserver/authenticator/patch_authenticator.go#L98))
+     to distinguish our "human" users from users authenticated otherwise, so that these
+     are capable of creating their own projects. Would it be ok to add this group to
+     anyone authenticated via a webhookTokenAuthenticator?
+
+## Summary
+
+`webhookTokenAuthenticators` configuration of the kube-apiserver will be configurable and will be leveraged by
```

Comment:
may something more like

> The standard mechanism for external authentication integration is via `tokenreviews.authentication.k8s.io` 
from the kube-apiserver.  We will support configuring the kube-apiserver this way and ensure it always
works by using the same mechanism to drive the standard openshift oauth-server authentication integration.